pkg/engine/turn/modify.go

```diff
@@ -1,48 +1,64 @@
 package turn
 
 import (
+	"sort"
 	"github.com/simimpact/srsim/pkg/engine/event"
 	"github.com/simimpact/srsim/pkg/engine/info"
 	"github.com/simimpact/srsim/pkg/key"
 )
 
+// SetGauge sets the gauge of the target as detailed in the input ModifyAttribute struct.
+// 1. Update gauge of target
+// 2. Move target to top of order
+// 3. Sort target down based on AV. In the event of tie, this target should be at top of order.
+//		If there is an active turn and it is not this target, this target should go below the active
+//		turn (so index 1 instead of 0 when 0 gauge/AV)
+// 4. Emit GaugeChangeEvent
 func (mgr *manager) SetGauge(data info.ModifyAttribute) error {
-	prev := mgr.target(data.Target).gauge
+
+	previousGauge := mgr.target(data.Target).gauge
 	mgr.target(data.Target).gauge = data.Amount
 
-	if mgr.target(data.Target).gauge == prev {
+	// if there's no change to Gauge, exit early
+	if mgr.target(data.Target).gauge == previousGauge {
 		return nil
 	}
 
-	// create map of TargetID -> AV so you only need to calc once within this call
-	targetAV := make(map[key.TargetID]float64, len(mgr.order.order))
-
-	// TODO:
-	// 1. update gauge of target
-	// 2. move target to top of order
-	// 3. sort target down based on AV. In the event of tie, this target should be at top of order.
-	//		If there is an active turn and it is not this target, this target should go below the active
-	//		turn (so index 1 instead of 0 when 0 gauge/AV)
-	// 4. emit GaugeChangeEvent
+	// find target index in mgr.orderHandler.turnOrder
+	targetIndex := 0
+	for i, v := range mgr.orderHandler.turnOrder {
+		if data.Target == v.id {
+			targetIndex = i
+			break
+		}
+	}
```

Comment:
move this to a function to cleanup usage. Same index discover exists in `RemoveTarget`. May want to update `Target(key.TargetID)` to return both the index and `*Target` since it is also a similar loop?

Should also account for case where target is not found and throw an error (here and `RemoveTarget`)